processor/queuedprocessor/queued_processor.go

```diff
@@ -109,27 +113,27 @@ func (sp *queuedSpanProcessor) Stop() {
 }
 
 // ConsumeTraceData implements the SpanProcessor interface
-func (sp *queuedSpanProcessor) ConsumeTraceData(ctx context.Context, td consumerdata.TraceData) error {
+func (sp *queuedSpanProcessor) ConsumeTrace(ctx context.Context, td data.TraceData) error {
 	ctx = obsreport.ProcessorContext(ctx, sp.name)
 	item := &queueItem{
 		queuedTime: time.Now(),
 		td:         td,
 		ctx:        ctx,
 	}
 
-	statsTags := processor.StatsTagsForBatch(sp.name, processor.ServiceNameForNode(td.Node), td.SourceFormat)
-	numSpans := len(td.Spans)
-	stats.RecordWithTags(ctx, statsTags, processor.StatReceivedSpanCount.M(int64(numSpans)))
+	statsTags := processor.StatsTagsForBatch(sp.name, processor.ServiceNameForTraceData(td))
+	spanCount := item.td.SpanCount()
+	stats.RecordWithTags(ctx, statsTags, processor.StatReceivedSpanCount.M(int64(spanCount)))
 
 	addedToQueue := sp.queue.Produce(item)
 	if !addedToQueue {
 		// TODO: in principle this may not end in data loss because this can be
 		// in the same call stack as the receiver, ie.: the call from the receiver
 		// to here is synchronous. This means that actually it could be proper to
 		// record this as "refused" instead of "dropped".
-		sp.onItemDropped(item, statsTags)
+		sp.onItemDropped(item, statsTags, spanCount)
 	} else {
-		obsreport.ProcessorTraceDataAccepted(ctx, td)
+		obsreport.ProcessorTraceDataAccepted(ctx, spanCount)
```

Comment:
the calls to obsreport can be combined 2 of each. Maybe add a todo?

Reply:
Sorry, I didn't get it. Which calls can be combined?